protocols/request-response/src/lib.rs

```diff
@@ -470,20 +470,14 @@ where
     ///
     /// Returns true if the address was added, false otherwise (i.e. if the
     /// address is already in the list).
+    #[deprecated(note = "Use `libp2p_swarm::Swarm::add_peer_address` instead.")]
     pub fn add_address(&mut self, peer: &PeerId, address: Multiaddr) -> bool {
-        self.addresses.entry(*peer).or_default().insert(address)
+        self.addresses.put(*peer, std::iter::once(address))
```

Comment:
This currently **overwrites** all addresses. I don't think that is what we want.

Reply:
Changed the internals of `put`: https://github.com/libp2p/rust-libp2p/pull/4371/files#diff-cc3b2f08e48cc81069062816551b4be65ecce092128bca640d5d5024b6e9d6beR50